lib/src/model/tile_model.dart

```diff
@@ -35,34 +39,44 @@ class TileFeature {
     }
 
     final geometry = _geometry;
-    if (geometry != null) {
+    if (geometry != null && _points == null) {
       _points = decodePoints(geometry).toList(growable: false);
-      _geometry = null;
     }
 
-    return _points;
+    return _points ?? [];
   }
 
-  List<Path> get paths {
+  List<Path> getPaths({List<num> dashLengths = const []}) {
     if (type == TileFeatureType.point) {
       throw StateError('Cannot get paths from a point feature');
     }
 
     final geometry = _geometry;
     if (geometry != null) {
-      // ignore: missing_enum_constant_in_switch
-      switch (type) {
-        case TileFeatureType.linestring:
-          _paths = decodeLineStrings(geometry).toList(growable: false);
-          break;
-        case TileFeatureType.polygon:
-          _paths = decodePolygons(geometry).toList(growable: false);
-          break;
+      if (_paths == null) {
+        // ignore: missing_enum_constant_in_switch
+        switch (type) {
+          case TileFeatureType.linestring:
+            _paths = decodeLineStrings(geometry).toList(growable: false);
+            break;
+          case TileFeatureType.polygon:
+            _paths = decodePolygons(geometry).toList(growable: false);
+            break;
+        }
       }
-      _geometry = null;
     }
 
-    return _paths;
+    if (dashLengths.length >= 2) {
+      if (dashedPaths.containsKey(dashLengths)) {
+        return dashedPaths[dashLengths]!;
+      } else {
+        final path = _paths?.map((e) => e.dashPath(RingNumberProvider(dashLengths))).toList() ?? [];
```

Comment:
I'm a bit confused. Why do we need to create a new list here?

Reply:
My thought was something like: A tilefeature might be repainted with different styles (eg. No dashing or different dashes). That means we need to cache all combinations in the map. Can this occur? 
What is exactly the lifecycle of thr tilefeature object and how long is it kept in memory?